crates/rpc-types-eth/src/state.rs

```diff
@@ -7,6 +7,71 @@ use alloy_primitives::{
     Address, Bytes, B256, U256,
 };
 
+/// A StateOverride builder.
+#[derive(Clone, Debug, Default, PartialEq, Eq)]
+pub struct StateOverridesBuilder {
+    overrides: StateOverride,
+}
+
+impl StateOverridesBuilder {
+    /// Create a new StateOverridesBuilder.
+    pub fn new(map: AddressHashMap<AccountOverride>) -> Self {
+        Self { overrides: map }
+    }
+
+    /// Adds an account override for a specific address.
+    pub fn insert(mut self, address: Address, account_override: AccountOverride) -> Self {
+        self.overrides.insert(address, account_override);
+        self
+    }
```

Comment:
I'd like to rename this to append, and then we also need `fn extend` that accepts in impl intoiter